std.outbuffer: remove gratuitious calls into the GC

rainers opened the pull request:
- using appropriate type for storage array data allows removing GC.clrAttr(NO_SCAN) calls
- pre-allocating a little more data avoids a number of GC calls that actually don't change the allocation size

DmitryOlshansky commented:
Looks good but I'd suggest to wrap cast(ubyte[])data as internal property function. May call it "raw" or some such.

rainers commented:
> I'd suggest to wrap cast(ubyte[])data as internal property function.

I thought about that, too, but wondered whether this could have a performance impact for debug builds. Unfortunately, `data` is public, so I might have to convert it into a property instead.

rainers commented:
> but wondered whether this could have a performance impact for debug builds.

Being a class with invariants, performance is probably not a requirement for debug builds here. I changed `data` to be a `ubyte[]` property and renamed the actual storage to `vdata`.

DmitryOlshansky commented:
LGTM now

rainers commented:
Thinking about it again, I think it's less invasive to just change the type of the allocation. Updated to that respect.

DmitryOlshansky commented:
Even better - maybe move "allocate as void" comment to vdata.length line? Also would it really clear no_scan bit?

rainers commented:
> Also would it really clear no_scan bit?

The relevant code lines are https://github.com/D-Programming-Language/druntime/blob/master/src/rt/lifetime.d#L415 and https://github.com/D-Programming-Language/druntime/blob/master/src/rt/typeinfo/ti_void.d#L63.

rainers commented:
> maybe move "allocate as void" comment to vdata.length line?

Ok, done. Probably clearer about where the allocation actually happens.
